extensions/src/poseHand/index.ts

```diff
@@ -0,0 +1,324 @@
+    this.runtime.emit(this.runtime.constructor.PERIPHERAL_CONNECTED);
+    */
+
+    this.firstInstall = true;
+
+    if (this.runtime.ioDevices) {
+      /* Possibly unnecessary, keep commented just in case
+      this.runtime.on(RuntimeEvent.ProjectLoaded, this.projectStarted.bind(this));
+      this.runtime.on(RuntimeEvent.ProjectRunStart, this.reset.bind(this)); 
+      */
+      this._loop();
+    }
+  }
+
+  /**
+   * Dimensions the video stream is analyzed at after its rendered to the
+   * sample canvas.
+   * @type {Array.<number>}
+   */
+  static get DIMENSIONS() {
+    return [480, 360];
+  }
+
+  /**
+   * Converts the coordinates from the hand pose estimate to Scratch coordinates
+   * @param x 
+   * @param y
+   * @param z
+   * @returns enum
+   */
+  tfCoordsToScratch({ x, y, z }) {
+    return { x: x - 250, y: 200 - y };
+  }
+
+  /**
+   * Get the latest values for video transparency and state,
+   * and set the video device to use them.
+   */
+  projectStarted() {
+    this.setVideoTransparency(this.globalVideoTransparency);
+    this.videoToggle(this.globalVideoState);
+  }
+
+  /**
+   * init() binds to this function, but it is never called, so this may be unimportant
+   */
+  // reset() {
+  // }
+
+  /**
+   * Checks if the hand pose estimate is ready to be used
+   * @returns {boolean} true if connected, false if not connected
+   */
+  isConnected() {
+    return !!this.handPoseState && this.handPoseState.length > 0;
+  }
+
+  /**
+   * Runs for the entire time the extension is running. Gets information about the video frame.
+   * Estimates where the hand is on the video frame. Creates a delay to prevent this function from constantly running,
+   * so as to prevent the entire program from slowing down.
+   */
+  async _loop() {
+    while (true) {
+      const frame = this.runtime.ioDevices.video.getFrame({
+        format: Video.FORMAT_IMAGE_DATA,
+        dimensions: PoseHand.DIMENSIONS
+      });
+
+      const time = +new Date();
+      if (frame) {
+        this.handPoseState = await this.estimateHandPoseOnImage(frame);
+        /*
+        if (this.isConnected()) {
+            this.runtime.emit(this.runtime.constructor.PERIPHERAL_CONNECTED);
+        } else {
+            this.runtime.emit(this.runtime.constructor.PERIPHERAL_DISCONNECTED);
+        }
+        */
+      }
+      const estimateThrottleTimeout = (+new Date() - time) / 4;
+      await new Promise(r => setTimeout(r, estimateThrottleTimeout));
+    }
+  }
+
+  /**
+   * Estimates where the hand is on the video frame.
+   * @param imageElement
+   * @returns {Promise<AnnotatedPrediction[]>}
+   */
+  async estimateHandPoseOnImage(imageElement) {
+    const handModel = await this.getLoadedHandModel();
+    return await handModel.estimateHands(imageElement, {
+      flipHorizontal: false
+    });
+  }
+
+  /**
+   * Gets the hand model from handpose
+   * @returns
+   */
+  async getLoadedHandModel() {
+    if (!this._handModel) {
```

Comment:
Something that could be used here is [nullish coalescing assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_assignment), though I'm not totally sure how it behaves with async await -- might be fine!

This would become:
```
this._handModel ??= await handpose.load();
return this._handModel;
```